packages/calcite-components/src/components/filter/filter.tsx

```diff
@@ -169,6 +169,7 @@ export class Filter
   disconnectedCallback(): void {
     disconnectLocalized(this);
     disconnectMessages(this);
+    this.filterDebounced.cancel();
```

Comment:
Nice! 

Would you need to filter again on `connectedCallback`? Might be an edge case, but I'm thinking of the case where filtering is running as the component is moved between parents.

Reply:
Maybe we could address that if it ever comes up? Since we don't know if a component is moved vs removed we would have to maintain state like this that isn't set via a prop. It would be something to discuss across all components that have some async method or state to maintain.

Reply:
For sure, we can keep an eye out for it. I think this is a very edge use case anyways since the filter is a supporting component and it'd be odd for a component to move it around during filtering. 🤔